pkg/resource/hyper_parameter_tuning_job/custom_set_output.go

```diff
@@ -0,0 +1,93 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Use this file if the Status/Spec of the CR needs to be modified after
+// create/describe/update operation
+
+package hyper_parameter_tuning_job
+
+import (
+	"context"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	svcapitypes "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// customCreateHyperParameterTuningJobSetOutput sets the resource in TempOutofSync if HyperParameterTuningJob is
+// in creating state. At this stage we know call to createEndpoint was successful.
+func (rm *resourceManager) customCreateHyperParameterTuningJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.CreateHyperParameterTuningJobOutput,
+	ko *svcapitypes.HyperParameterTuningJob,
+) (*svcapitypes.HyperParameterTuningJob, error) {
+	rm.customSetOutput(r, aws.String(svcsdk.HyperParameterTuningJobStatusInProgress), ko)
+	return ko, nil
+}
+
+// customStopHyperParameterTuningJobSetOutput sets the resource in TempOutofSync if HyperParameterTuningJob is
+// in creating state. At this stage we know call to createEndpoint was successful.
+func (rm *resourceManager) customStopHyperParameterTuningJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.StopHyperParameterTuningJobOutput,
+	ko *svcapitypes.HyperParameterTuningJob,
+) (*svcapitypes.HyperParameterTuningJob, error) {
+	rm.customSetOutput(r, aws.String(svcsdk.HyperParameterTuningJobStatusStopping), ko)
+	return ko, nil
+}
+
+// customSetOutput sets ConditionTypeResourceSynced condition to True or False
+// based on the hyperParameterTuningJobStatus on AWS so the reconciler can determine if a
+// requeue is needed
+func (rm *resourceManager) customSetOutput(
+	r *resource,
+	hyperParameterTuningJobStatus *string,
+	ko *svcapitypes.HyperParameterTuningJob,
+) {
+	if hyperParameterTuningJobStatus == nil {
+		return
+	}
+
+	syncConditionStatus := corev1.ConditionUnknown
+	if *hyperParameterTuningJobStatus == svcsdk.HyperParameterTuningJobStatusCompleted || *hyperParameterTuningJobStatus == svcsdk.HyperParameterTuningJobStatusStopped {
+		syncConditionStatus = corev1.ConditionTrue
+		syncConditionStatus = corev1.ConditionFalse
```

Comment:
should be one or the other, right? :)

Reply:
That's a copy paste error, but good catch!